src/main/java/de/qabel/desktop/daemon/drop/DropDaemon.java

```diff
@@ -55,21 +57,20 @@ public void run() {
 		}
 	}
 
-	void receiveMessages() throws PersistenceException {
+	void receiveMessages() throws PersistenceException, EntityNotFoundExcepion {
 		Identity identity = config.getSelectedIdentity();
-		java.util.List<DropMessage> dropMessages = httpDropConnector.receive(identity, lastDate);
-
+		java.util.List<DropMessage> dropMessages;
+		try {
+			dropMessages = httpDropConnector.receive(identity, lastDate);
+		} catch (NullPointerException e){
+			return;
+		}
 		for (DropMessage d : dropMessages) {
-			Contact contact = null;
-			try {
-				contact = contactRepository.findByKeyId(identity, d.getSenderKeyId());
-			} catch (EntityNotFoundExcepion e) {
-				logger.error("Contact: with ID: " + d.getSenderKeyId() + " not found " + e.getMessage(), e);
-				continue;
-			}
+
 			lastDate = config.getLastDropPoll(identity);
 			if (lastDate.getTime() < d.getCreationDate().getTime()) {
-				dropMessageRepository.addMessage(d, contact, false);
+				Contact sender = contactRepository.findByKeyId(identity, d.getSenderKeyId());
```

Comment:
this throws the EntityNotFoundException ... if you get a message from an unknown contact, you won't receive any further messages.